### PR TITLE
Fix structure service for implicit module

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This fixes https://github.com/Microsoft/visualfsharp/issues/2908

### CODE REVIEW

File: src/fsharp/vs/ServiceStructure.fs [hunk 1]
```diff
@@ -609,33 +605,35 @@ module internal Structure =
...
-    let private parseModuleOrNamespace (SynModuleOrNamespace.SynModuleOrNamespace (longId,_,isModule,decls,_,attribs,_,r)) =
+    let parseModuleOrNamespace isScript (SynModuleOrNamespace (longId,_,isModule,decls,_,attribs,_,r)) =
         seq {
             yield! parseAttributes attribs
             let idRange = longIdentRange longId
             let fullrange = Range.startToEnd idRange r  
             let collapse = Range.endToEnd idRange r 
-            if isModule then
+            
+            // do not return range for top level implicit module in scripts
```
Comment (saul) [inline src/fsharp/vs/ServiceStructure.fs]: Are we sure that this doesn't affect modules within scripts?

Comment (vasily-kirichenko) [inline src/fsharp/vs/ServiceStructure.fs]: Yes, nested modules are handled by a different function.

**Comment by cartermp** [comment]
Perhaps this is just my unfamiliarity, but it seems like this overall feature is now calling both into `Structure.fs` in FSharp.Editor and the new file that you've created with this PR.  Is there an opportunity to merge those together, and perhaps even with `ServiceUntypedParse.fs`?

**Comment by vasily-kirichenko** [comment]
Structure.fs was moved to fsc and renamed to ServiceStructure.fs + ServiceStructure.fsi

**Comment by KevinRansom** [comment]
thank you

**Pull request merged**